front/components/vaults/VaultResourcesList.tsx

```diff
@@ -367,9 +402,16 @@ export const VaultResourcesList = ({
                 owner={owner}
                 dustClientFacingUrl={dustClientFacingUrl}
                 plan={plan}
-                existingDataSources={vaultDataSourceViews.map(
-                  (v) => v.dataSource
-                )}
+                existingDataSources={
+                  vaultDataSourceViews
+                    .filter(
+                      (dsView) => dsView.dataSource.connectorProvider !== null
+                    )
+                    .map(
+                      (v) => v.dataSource
+                    ) as DataSourceWithConnectorDetailsType[]
```

Comment:
Could we avoid casting here?

Reply:
Nope, otherwise it complains, it is not able to understand that we filtered out the managed ones (i spent 30 minutes trying to avoid the cast). Any better idea ?